### PR TITLE
Add REST API to list report and download pdf report

### LINKED ISSUE #3681: Create REST API to fetch reports
Create REST API to list all the reports and to download a report

### PR DESCRIPTION
**Pull request by dekkers**
### Changes

This adds a minimal REST API to list generated reports and download the pdf report. The PDF endpoint redirect to the normal PDF generation view. It would be possible to return the pdf in the API, but it looks like we would need to refactor a lot for that. I think that if we want that we should do this later.

The organization can be giving with either `organization_id` (the primary key in the rocky database) or `organization_code` query parameter. The valid time can be given with the `valid_time` parameter and default to the current time.

Unit tests are still missing.

### Issue link

Closes #3681 

### Demo

```
% http --body GET 'http://127.0.0.1:8000/api/v1/report/?organization_id=1'  "Authorization:Token ..."
{
    "count": 2,
    "next": null,
    "previous": null,
    "results": [
        {
            "generated_at": "2024-10-15T13:36:26.383897Z",
            "id": "dc6d9a09-c9b4-4f22-8501-e3bbe2b1f804",
            "intput_oois": [
                "Hostname|internet|mispo.es",
                "IPAddressV4|internet|134.209.85.72"
            ],
            "name": "Concatenated Report for 2 objects",
            "report_type": "concatenated-report",
            "valid_time": "2024-10-15T23:59:59.999999Z"
        },
        {
            "generated_at": "2024-10-15T11:46:15.436142Z",
            "id": "66a6ffc2-2e58-4457-877c-8315b1ab438c",
            "intput_oois": [
                "Hostname|internet|mispo.es",
                "IPAddressV4|internet|134.209.85.72"
            ],
            "name": "Aggregate Report",
            "report_type": "aggregate-organisation-report",
            "valid_time": "2024-10-15T11:46:14.829684Z"
        }
    ]
}
```

```
% http --body GET 'http://127.0.0.1:8000/api/v1/report/dc6d9a09-c9b4-4f22-8501-e3bbe2b1f804/?organization_id=1'  "Authorization:Token ..."
{
    "generated_at": "2024-10-15T13:36:26.383897Z",
    "id": "dc6d9a09-c9b4-4f22-8501-e3bbe2b1f804",
    "intput_oois": [
        "Hostname|internet|mispo.es",
        "IPAddressV4|internet|134.209.85.72"
    ],
    "name": "Concatenated Report for 2 objects",
    "report_type": "concatenated-report",
    "valid_time": "2024-10-15T23:59:59.999999Z"
}
```

```
% http GET 'http://127.0.0.1:8000/api/v1/report/dc6d9a09-c9b4-4f22-8501-e3bbe2b1f804/pdf/?organization_id=1'  "Authorization:Token ..." 
HTTP/1.1 302 Found
Allow: GET, HEAD, OPTIONS
Content-Language: en
Content-Length: 0
Content-Security-Policy: img-src 'self'; connect-src 'self'; frame-ancestors 'none'; base-uri 'none'; style-src 'self'; font-src 'self'; default-src 'none'; form-action 'self'; bloc
k-all-mixed-content
Content-Type: text/html; charset=utf-8
Cross-Origin-Opener-Policy: same-origin
Date: Tue, 15 Oct 2024 15:50:10 GMT
Location: /en/test/reports/view/pdf/?report_id=Report%7Cdc6d9a09-c9b4-4f22-8501-e3bbe2b1f804&observed_at=2024-10-15T15%3A50%3A10.717880%2B00%3A00
Referrer-Policy: same-origin
Server: WSGIServer/0.2 CPython/3.11.10
Vary: Accept, Accept-Language, Cookie
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
```

### QA notes

See demo

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by underdarknl** [comment]
Do the same login credentials work on the subsequent redirected pdf location?

**Comment by dekkers** [comment]
> Do the same login credentials work on the subsequent redirected pdf location?

Yes, we already implemented earlier that the token auth also works for the normal views.

File: rocky/account/mixins.py [hunk 1]
```diff
@@ -203,3 +206,50 @@ class OrganizationPermissionRequiredMixin(PermissionRequiredMixin):
...
+
+    @cached_property
+    def organization(self) -> Organization:
+        try:
+            organization_id = self.request.query_params["organization_id"]
+        except KeyError:
+            pass
+        else:
+            try:
+                return Organization.objects.get(id=organization_id)
```
Comment (underdarknl) [inline rocky/account/mixins.py:222]: I'd rather have this factory as a method on the user object, where the user instance limits the organizations visible/loadbable to it by checking its permissions and memberships.

**Review by underdarknl — COMMENTED** [review]
Some ideas, maybe not mvp

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Followed the QA notes and works as expected. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**